Configuration/EventContent/python/EventContent_cff.py

```diff
@@ -620,6 +620,7 @@ def SwapKeepAndDrop(l):
 FEVTDEBUGEventContent.outputCommands.extend(SimMuonFEVTDEBUG.outputCommands)
 FEVTDEBUGEventContent.outputCommands.extend(SimCalorimetryFEVTDEBUG.outputCommands)
 FEVTDEBUGEventContent.outputCommands.extend(SimFastTimingFEVTDEBUG.outputCommands)
+FEVTDEBUGEventContent.outputCommands.append('keep recoMuons_muons1stStep_*_*')
```

Comment:
```suggestion
 phase2_muon.toModify(FEVTDEBUGHLTEventContent, outputCommands = FEVTDEBUGHLTEventContent.outputCommands + ['keep recoMuons_muons1stStep_*_*'])
```
@srimanob @mandrenguyen is it better to have something like that? In this way it should only affect Phase-2 workflows.

Reply:
Wouldn't a more elegant solution be to rename muons to remuons or something in the re-TICL workflow. I think we can use the MassReplaceInputTag functionality

Reply:
If I understand correctly, the only reason that you cannot use `muons` as input is because you are reproducing this same collection after TICL.  Is that correct?  If in the TICL step you renamed the collection you would avoid this problem without increasing the FEVT event size.

Reply:
Yes, but let me summarize the problem: in TICL we consume `recoMuons_muons1stStep` and not `recoMuons_muons` because `recoMuons_muons` are produced after TICL. But the `recoMuons_muons1stStep` are not saved in the Event Content. We want to have the very same version of TICL in the re-TICL framework, and that's why we want to use `recoMuons_muons1stStep` in re-TICL and that's why we would need them in the Event Content.

Reply:
Are `muons1stStep` and `muons` actually functionally different from the point of view of TICL?  In other words, if you ran TICL with `muons` as input, would you actually get a different result?

Reply:
Hi @mandrenguyen , in principle yes, we would get different results. On top of that I think it would be very weird and risky to have a "hidden" circular dependency in TICLFromReco

Reply:
Do you know what exactly the difference between `muons` and `muons1stStep` is?

Reply:
Sorry to jump in.
TICL cannot consume `muons` because that introduces a circular dependency, as already verified by @waredjeb 
The amount of space we are asking for (limited to Phase2 samples) in the `FEVTDEBHLT` data-tier is quite minimal if compared to its current size. 
Moreover, IIRC, the FEVTDEBUGHLT is mostly meant for debugging, not for big, central productions.
Is there any fundamental reason to hold this PR any further?

Reply:
Yes, I already understood the circular dependency. 
What is the fractional increase in `FEVTDEBUGHLT`?
Isn't all MC production writing this heavy format?

Reply:
I'm not sure my question was answered as well.  Is there a functional difference between `muons` and `muons1stStep` in practice?

Reply:
> Yes, I already understood the circular dependency.
> What is the fractional increase in `FEVTDEBUGHLT`?
> Isn't all MC production writing this heavy format?

I'm sure @waredjeb can also provide the fractional increase.

MC productions for Phase 2 are still to be considered mostly for development (and debugging). Having said that, I'm not entirely sure all centrally produced samples are saved in FEVTDEBUHLT format. Maybe Upgrade Software L2 could shed some light on this point.
 
As for your last question, I'd say Muon POG could provide all the gory details about that.

Reply:
MC production for Phase-2 mostly produces GEN-SIM-DIGI-RAW-MINIAOD (which is FEVTDEBUGHLT (up to HLT step) + MiniAOD). Size is ~70M /event. We don't keep FEVTDEBUGHLT (up to RECO step), except request.

To me, if you use `phase2_muon.toModify` is OK, so it will change only Phase-2.